src/imageio/format/jxl.c

```diff
@@ -0,0 +1,703 @@
+{
+  // By default error code
+  int out = 1;
+
+  const float *in = (float *)in_tmp;
+  float *pixels = NULL;
+  uint8_t *out_buf = NULL;
+  FILE *out_file = NULL;
+
+#define LIBJXL_ASSERT(code)                                                                                       \
+  {                                                                                                               \
+    const JxlEncoderStatus res = code;                                                                            \
+    if(res != JXL_ENC_SUCCESS)                                                                                    \
+    {                                                                                                             \
+      fprintf(stderr, "JXL: libjxl call failed with err %d (src/imageio/format/jxl.c#L%d)\n", res, __LINE__);     \
+      goto end;                                                                                                   \
+    }                                                                                                             \
+  }
+
+#define JXL_FAIL(msg, ...)                                                                                        \
+  {                                                                                                               \
+    fprintf(stderr, "JXL: " msg "\n", ##__VA_ARGS__);                                                             \
+    goto end;                                                                                                     \
+  }
+
+  const dt_imageio_jxl_params_t *params = (dt_imageio_jxl_params_t *)data;
+  const int width = params->global.width;
+  const int height = params->global.height;
+
+  JxlEncoder *encoder = JxlEncoderCreate(NULL);
+
+  // JxlThreadParallelRunnerDefaultNumWorkerThreads()
+  const uint32_t num_threads = JxlResizableParallelRunnerSuggestThreads(width, height);
+  void *runner = JxlThreadParallelRunnerCreate(NULL, num_threads);
+  LIBJXL_ASSERT(JxlEncoderSetParallelRunner(encoder, JxlThreadParallelRunner, runner));
+
+  const dt_colorspaces_color_profile_t *output_profile
+      = dt_colorspaces_get_output_profile(imgid, over_type, over_filename);
+  const cmsHPROFILE out_profile = output_profile->profile;
+  // Previous call will give us a more accurate color profile type
+  // (not what the user requested in the export menu but what the image is actually using)
+  over_type = output_profile->type;
+
+  // If possible we want libjxl to save the color encoding in it's own format, rather
+  // than as an ICC binary blob which is possible.
+  // If we are unable to find the required color encoding data for libjxl we will
+  // just fallback to providing an ICC blob (and hope we can at least do that!).
+  bool written_color = false;
+
+  // We can only handle RGB
+  if(cmsGetColorSpace(out_profile) == cmsSigRgbData)
```

Comment:
Could even remove this check because [all dt profiles are RGB currently](https://github.com/darktable-org/darktable/blob/80839f97d898c213edeecb543d46deb6a6f48452/src/common/colorspaces.c#L987)...